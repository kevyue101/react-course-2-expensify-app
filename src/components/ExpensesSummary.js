import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => {
    const expenseWord = props.expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(props.expensesTotal/100).format('$0,0.00');
    return (
    <div>
        <div className="page-header">
            <div className="content-container">
                {
                    props.expensesCount === 0 ? 
                    (
                        <h1>No expenses to add up</h1>
                    ) : 
                    (
                        <h1 className="page-header__title">
                        Viewing <span>{props.expensesCount}</span> <span>{expenseWord}</span> totalling <span>{formattedExpensesTotal}</span>
                        </h1>
                    )
                }
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
        {
  
        }      
    </div>
)};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
