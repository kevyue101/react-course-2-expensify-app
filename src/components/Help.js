import React from 'react';

const HelpPage = () => (
    <div>
        This is from my help comp
    </div>
);

export default HelpPage;