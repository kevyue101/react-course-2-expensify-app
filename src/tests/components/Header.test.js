import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let wrapper, startLogout;

test('should render Header correctly', () => {
    wrapper = shallow(<Header startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should log out when button is clicked', () => {
    startLogout = jest.fn();
    wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find('button').at(0).simulate('click');
    expect(startLogout).toHaveBeenCalled();
});