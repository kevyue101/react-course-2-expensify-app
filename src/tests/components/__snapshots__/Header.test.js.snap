// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <div>
    <h3>
      Expensify
    </h3>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/dashboard"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
      Create Expense
    </NavLink>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
